bookshelf/app/controllers/registrations_controller.rb

```diff
@@ -0,0 +1,51 @@
+# Talentica - bootstrap project
+# Copyright   Ashwini Kumar
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+class RegistrationsController < Devise::RegistrationsController
+  # registration of new user by overriding native devise
+  before_filter :authenticate_user!
+
+  def email_availabilty
+    @user = User.has_email_as params["email"]
+    @response = if @user
+                  "Email already exist"
+                else
+                  "Email doesn't exist with us"
+                end
+    respond_to do |format|
+      format.json { render json: @response }
```

Comment:
Instead of plain string, the return json should be a valid json object, such as:
```javascript
{
  "available": false,
  "message": "The Email id is already taken"
}
```